label_studio/projects/functions/utils.py

```diff
@@ -26,3 +26,17 @@ def make_queryset_from_iterable(tasks_list):
                 raise ValueError(f'Unknown object type: {str(task)}')
         queryset = Task.objects.filter(id__in=ids)
     return queryset
+
+
+def recalculate_created_labels_from_scratch(project, summary):
+    summary.created_labels = {}
+    if project.annotations.exists():
+        summary.update_created_annotations_and_labels(project.annotations.all())
+    else:
+        summary.save(update_fields=['created_labels'])
+    summary.created_labels_drafts = {}
+    drafts = AnnotationDraft.objects.filter(task__project=project)
+    if drafts.exists():
+        summary.update_created_labels_drafts(drafts)
+    else:
+        summary.save(update_fields=['created_labels_drafts'])
```

Comment:
Do we really need this branching? Seems to me that if there are no drafts, `update_created_labels_drafts` should save an empty dict of drafts anyway, right?

Reply:
I think the invariant that this call resets to `{}` if there are no drafts would be better captured in a unit test instead (same for https://github.com/HumanSignal/label-studio/pull/5432#discussion_r1480632602)

Reply:
hm.. sorry, I don't understand what invariant you mean..

Reply:
The behavior where there are no drafts on the project is now captured by this unit test:

https://github.com/HumanSignal/label-studio/pull/5432/files#diff-fc2a039da0975ee919925c1f66e53984db5044b15d741eed50f66837eae03364R63-R95

The behavior where there are no annotations on the project is now captured by this unit test:

https://github.com/HumanSignal/label-studio/pull/5432/files#diff-fc2a039da0975ee919925c1f66e53984db5044b15d741eed50f66837eae03364R27-R60

Now we don't have to check whether `drafts.exists()` in this function, because we know from the tests that the `update_created_labels_drafts` function will do the right thing when the field on the summary has been cleared.